From c20027257b2c3938ebf84a18e93da7bc9008907d Mon Sep 17 00:00:00 2001
From: Nate Myren <ntmyren@google.com>
Date: Tue, 6 Dec 2022 14:01:03 -0800
Subject: [PATCH] RESTRICT AUTOMERGE Finish ManagePermissionsActivity if device
 is not provisioned

If the device isn't set up yet, do not allow access to the permissions
settings

Bug: 253043490
Test: manual
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:5e297ab51388db5375093f7dc21d37bd59de827c)
Merged-In: I6e8fb8f2d934cff965069493740cfc1c59c3623f
Change-Id: I6e8fb8f2d934cff965069493740cfc1c59c3623f
---
 .../permission/ui/ManagePermissionsActivity.java       | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/PermissionController/src/com/android/permissioncontroller/permission/ui/ManagePermissionsActivity.java b/PermissionController/src/com/android/permissioncontroller/permission/ui/ManagePermissionsActivity.java
index bf48e28ae..28e0d23bb 100644
--- a/PermissionController/src/com/android/permissioncontroller/permission/ui/ManagePermissionsActivity.java
+++ b/PermissionController/src/com/android/permissioncontroller/permission/ui/ManagePermissionsActivity.java
@@ -35,6 +35,7 @@ import android.content.pm.PermissionInfo;
 import android.os.Build;
 import android.os.Bundle;
 import android.os.UserHandle;
+import android.provider.Settings;
 import android.util.Log;
 import android.view.MenuItem;
 
@@ -122,6 +123,15 @@ public final class ManagePermissionsActivity extends SettingsActivity {
             return;
         }
 
+        boolean provisioned = Settings.Global.getInt(
+                getContentResolver(), Settings.Global.DEVICE_PROVISIONED, 0) != 0;
+        boolean completed = Settings.Secure.getInt(
+                getContentResolver(), Settings.Secure.USER_SETUP_COMPLETE, 0) != 0;
+        if (!provisioned || !completed) {
+            finishAfterTransition();
+            return;
+        }
+
         android.app.Fragment fragment = null;
         Fragment androidXFragment = null;
         String action = getIntent().getAction();
-- 
2.41.0.255.g8b1d071c50-goog

